subxt/src/backend/rpc/rpc_client_t.rs

```diff
@@ -101,3 +101,94 @@ impl<T: RpcClientT> RpcClientT for Box<T> {
         (**self).subscribe_raw(sub, params, unsub)
     }
 }
+
+#[cfg(feature = "reconnecting-rpc-client")]
+impl RpcClientT for reconnecting_jsonrpsee_ws_client::Client {
```

Comment:
I would probably place this under a `src/client/reconnecting_client` module. Maybe within a `ReconnectingClientRpC(reconnecting_jsonrpsee_ws_client::Client)`.

Ideally we would have a `ReconnectingClient<T: Config>` that has a `ReconnectingClient::builder` method and implements `OnlineClientT` and `OfflineClientT` traits.
Then we could also expose on the `ReconnectingClient` a `reconnect_count` method; and other methods of interest.
This would tidy up a bit the user API. Although the tricky part would be that we'll duplicate the `reconnecting_jsonrpsee_ws_client::Client::builder` methods on the `ReconnectingClient::builder`.

Considering this is under a feature-flag at the moment, I'd leave that refactoring as a follow-up as probably would need some extra tought.
What do you think?

Reply:
My gut feeling is that for now it should just be a standalone optional RPC client that you can use to instantiate the existing `OnlineClient`, but at some point when we've added more special reconnect logic in places etc we can just use it by default in the current `OnlineClient` instead of the non-reconnecting-client :)

Reply:
I agree with the separate module though; something like `src/backend/rpc/reconnecting_jsonrpsee_impl.rs` maybe to sit next to the current `jsonrpsee_impl.rs`?